### PR TITLE
Add expense commands

### LINKED ISSUE #60: Add expense-related commands
To accomplish user stories #26 and #27.

Add the following commands:
- `add-expense`, alias `adde`
- `list-expense`, alias `lse`

### PR DESCRIPTION
**Pull request by ianyong**
Resolves #60.

Changes:
- Mirror #56.
- Resolve `TODO` in `ListIncomeCommandTest`
- Rename `expectedIncomeMultipleTags` to `expectedIncomeMultipleCategories`

### CODE REVIEW

**Review by yongping827 — CHANGES_REQUESTED** [review]

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/testutil/TransactionUtil.java [hunk 1]
```diff
@@ -26,6 +28,13 @@ public static String getAddCommand(Transaction transaction) {
         return AddCommand.COMMAND_WORD + " " + getTransactionDetails(transaction);
     }
 
+    /**
+     * Returns an add expense command string for adding the {@code transaction}.
```
Comment (wltan) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/testutil/TransactionUtil.java:32]: `transaction` is no longer the parameter name
```suggestion
     * Returns an add expense command string for adding the {@code expense}.
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/testutil/TransactionUtil.java [hunk 2]
```diff
@@ -26,6 +28,13 @@ public static String getAddCommand(Transaction transaction) {
...
+     */
+    public static String getAddExpenseCommand(Expense expense) {
+        return AddExpenseCommand.COMMAND_WORD + " " + getTransactionDetails(expense);
+    }
+
     /**
      * Returns an add income command string for adding the {@code transaction}.
```
Comment (wltan) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/testutil/TransactionUtil.java:39]: This should be changed as well, my bad!
```suggestion
     * Returns an add income command string for adding the {@code income}.
```

**Review by wltan — CHANGES_REQUESTED** [review]

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/AddExpenseCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.stream.Stream;
+
+import ay2021s1_cs2103_w16_3.finesse.logic.commands.AddExpenseCommand;
+import ay2021s1_cs2103_w16_3.finesse.logic.parser.exceptions.ParseException;
+import ay2021s1_cs2103_w16_3.finesse.model.category.Category;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Amount;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Date;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Expense;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Name;
+
```
[the remark this answers is not in the document]
Comment (ianyong) [reply src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/AddExpenseCommandParser.java:19]: @yongping827 The corresponding comment in `AddIncomeCommandParser` does not have a full stop, should I still add it?

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/testutil/TransactionUtil.java (hunk 1 above)
Comment (ianyong) [reply src/test/java/ay2021s1_cs2103_w16_3/finesse/testutil/TransactionUtil.java:32]: This was left as such to mirror the corresponding change in #56. Can I confirm this is not intended?

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/AddExpenseCommandParser.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (yongping827) [reply src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/AddExpenseCommandParser.java:19]: Add the full stop. The corresponding comment in `AddIncomeCommandParser` should have a full stop as well.

[the remark this answers is not in the document]
Comment (ianyong) [reply src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/AddExpenseCommandParser.java:19]: Full stops to be added separately. This is honestly very low priority.

**Review by yongping827 — APPROVED** [review]
LGTM

**Review by wltan — APPROVED** [review]
LGTM!

**Pull request merged**